openghg/store/base/_datasource.py

```diff
@@ -701,44 +700,41 @@ def save(self, bucket: str, new_version: bool = True) -> None:
                 self._data_keys[version_str_backup]["timestamp"] = str(timestamp_now())  # type: ignore
 
             # if not version_folder.exists():
-            version_folder.mkdir(parents=True, exist_ok=True)
+            # version_folder.mkdir(parents=True, exist_ok=True)
 
             # Iterate over the keys (daterange string) of the data dictionary
-            for daterange in self._data:
+            for daterange, data in self._data.items():
                 # data_key = f"{Datasource._data_root}/uuid/{self._uuid}/{version_str}/{daterange}"
                 data_key = self.define_data_key(label=daterange, version=version_str)
 
                 new_keys[daterange] = data_key
-                data = self._data[daterange]
-
-                filepath = version_folder / f"{daterange}._data"
+                # filepath = version_folder / f"{daterange}._data"
+                # TODO - we really just want to use set_object here but we need some bytes for that
+                # until the zarr changes are implemented let's stick to writing directly to NetCDF
+                # instead of writing to a temporary space and reading the bytes
+                filepath = Path(f"{bucket}/{data_key}._data")
 
                 try:
                     data.to_netcdf(filepath, engine="netcdf4")
                 except IOError:
-                    # If unable to write, return original data from back up.
-                    os.removedirs(version_folder)
-                    if delete_version:
-                        version_folder_backup.rename(version_folder)
-                        self._data_keys.pop(version_str_backup)
-                    raise Exception("Unable to write new data. Restored previous data.")
+                    try:
+                        filepath.parent.mkdir(parents=True)
+                        data.to_netcdf(filepath, engine="netcdf4")
+                    except IOError:
+                        # If unable to write, return original data from back up.
+                        # NOTE - I wasn't sure how the os.removedir() was restoring the backup here?
```

Comment:
This was removing the new `version_folder` and partial contents which had been created first and then restoring the original details from the back up (`version_folder_backup`). The version folder had to be deleted first because you can't do the equivalent of:
```$ mv version_folder_backup version_folder```
if version_folder contains anything.

Looking at the new workflow, it's likely we'd still need that step in the code.

Reply:
Correct me if I've missed it but I'm not seeing the equivalent of `os.removedirs(version_folder)` to delete the folder and any contained data before the backup data is restored? Just checking as I know you said this had been added?

Reply:
When we move data back to `version_key` the data at that key will get overwritten so we shouldn't need to delete it beforehand.

Reply:
As we discussed, this is related to the folder transfer problem. This is to cover the case where there's a been a problem creating the new version folder and we need to restore the backup. If we try to move a folder (e.g. version backup) to a non-empty folder (original version name) this will cause an error so to be safe we should delete what has been partially created.

Note that this is a fail case we're covering so it's to account for unexpected behaviour and to not leave the database in a half completed state.